src/components/ChooseSubmodule/SearchInput.tsx

```diff
@@ -0,0 +1,44 @@
+import { SearchIcon } from "@chakra-ui/icons";
+import {
+  InputGroup,
+  InputLeftElement,
+  Input,
+  forwardRef,
+} from "@chakra-ui/react";
+import { FormEventHandler, useEffect, useState } from "react";
+import { useDebounce } from "../../hooks/useDebounce";
+
+export interface SearchInputProps {
+  value: string;
+  onChange: (s: string) => void;
+  onSubmit: FormEventHandler<HTMLFormElement>;
+}
+
+export const SearchInput = forwardRef<SearchInputProps, "input">(
+  ({ value, onChange, onSubmit }, inputRef) => {
+    const [inputValue, setInputValue] = useState(value);
+    const debounced = useDebounce(inputValue, 250);
+
+    useEffect(() => {
+      onChange(debounced);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [debounced]);
```

Comment:
Im not sure if I understand this code correctly so please let me know if I'm a totally off. But I wonder if it would be good to have a `callback` function as an option parameter for `useDebounce`.

It would solve this linter problem and I feel like most of the times with `useDebounce` there is an action that will take place after the value has been resolved (if it's different).

Reply:
Not a bad idea, I've updated the hook's second arg to be an options object 👍